### PR TITLE
chore(ci): add semantic release

### PR DESCRIPTION
**Pull request by c0rydoras**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  semrel:
+    name: Semantic Release
+    runs-on: ubuntu-latest
+    permissions:
+      actions: none
+      checks: none
+      contents: none
+      deployments: none
+      issues: none
+      packages: write
```
Comment (hairmare) [inline .github/workflows/release.yaml:18]: i think it only needs `packages` if it also pushes an image

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      pull-requests: none
+      repository-projects: none
+      security-events: none
+      statuses: none
+      id-token: none
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```
Comment (hairmare) [inline .github/workflows/release.yaml:26]: does this fetch deep enough for semrel to word?

Comment (c0rydoras) [reply .github/workflows/release.yaml:26]: idk, it works in other repos

File: .github/workflows/release.yaml (hunk 1 above)
Comment (c0rydoras) [reply .github/workflows/release.yaml]: (updated it)

File: .github/workflows/release.yaml (hunk 2 above)
Comment (hairmare) [reply .github/workflows/release.yaml:26]: `fetch-depth` defaults to `1` to make it only fetch the last commit. this is often fine when analyzing 1 commit is enough to tag a release. if semrel needs to take multiple commit into account to figure out the new version then `fetch-depth: 0` will ensure that it has access to the full commit history.

This was for sure the case a while back and might have changed in newer versions of the checkout action. Do you know of an example repo that demonstrated it doing the right thing even when semrel doesn't have a complete picture in it's working copy?

Comment (hairmare) [reply .github/workflows/release.yaml:26]: lets try this as is then

**Pull request merged**